File: 0169/0169.ts
// Author: washing
// DateTime: 2021/1/26 18:48
// File: 0169.ts
// Desc: 


function majorityElement(nums: number[]): number {
    var a=0;
    var c=0;
    var n: number;
    for (var i=0;i<nums.length;i++){
        n = nums[i];
        if (c==0){
            a=n;
            c+=1;
        }else if(a==n){
            c+=1;
        }else{
            c-=1;
        }
    }
    return a;
}

File: 0169/0169.go
// Author: washing
// DateTime: 2021/1/26 18:51
// File: 0169.go
// Desc: 

func majorityElement(nums []int) int {
    var a, c, n int = 0, 0, 0
    for i := 0; i<len(nums); i++ {
        n = nums[i]
        if c == 0 {
            a = n
            c++
        }else if a == n {
            c++
        }else {
            c--
        }
    }
    return a
}